Wikipedia/Code/WMFImageGalleryViewController.m

```diff
@@ -409,7 +409,9 @@ - (UIView *_Nullable)photosViewController:(NYTPhotosViewController *)photosViewC
 
 - (void)updateImageForPhotoAfterUserInteractionIsFinished:(id<NYTPhoto> _Nullable)photo {
     //Exclude UITrackingRunLoopMode so the update doesn't happen while the user is pinching or scrolling
-    [self performSelector:@selector(updateImageForPhoto:) withObject:photo afterDelay:0 inModes:@[NSDefaultRunLoopMode]];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self performSelector:@selector(updateImageForPhoto:) withObject:photo afterDelay:0 inModes:@[NSDefaultRunLoopMode]];
```

Comment:
this could use `performSelector:onThread:` instead of having to async to the main thread